logic_mod/Blocks/Api/CpuMath.cs

```diff
@@ -114,9 +153,246 @@ public object Atan(VarCtx ctx, object[] x)
+        //      Tolerance (relative) for termination.
+        // full_output : bool, optional
+        //      If `full_output` is false (default), the root is returned.
+        //      If true, the dictionary {{"root": root}, {"converged": true/false},
+        //      {"iter": numIter}} is returned.
+        public object Newton(VarCtx ctx, object[] x)
+        {
+            int l = x.Length;
+
+            // Arguments and their default values:
+            object func;
+            float x0;
+            object fprime = null;
+            float tol = 1.48e-08F;
+            long maxiter = 50;
+            object fprime2 = null;
+            float x1 = 0;
+            float rtol = 0.0F;
+            bool full_output = false;
+
+            bool x1Provided = false;
+
+            if (l < 2 || l > 9)
+                throw new Exception("Invalid value");
+
+            // Conditionally initialize the arguments
+            int curArgIndex = 0;
+
+            func = x[curArgIndex++];
+
+            if (!BlockUtils.TryGetFloat(x[curArgIndex++], out x0))
+                throw new Exception("Invalid value");
+
+            if (curArgIndex >= l)
+                goto finishInitialization;
+            fprime = x[curArgIndex++];
+
+            if (curArgIndex >= l)
+                goto finishInitialization;
+            if (!BlockUtils.TryGetFloat(x[curArgIndex++], out tol))
+                throw new Exception("Invalid value");
+
+            if (curArgIndex >= l)
+                goto finishInitialization;
+            if (!BlockUtils.TryGetLong(x[curArgIndex++], out maxiter))
+                throw new Exception("Invalid value");
+
+            if (curArgIndex >= l)
+                goto finishInitialization;
+            fprime2 = x[curArgIndex++];
+
+
+            if (curArgIndex >= l)
+                goto finishInitialization;
+            x1Provided = BlockUtils.TryGetFloat(x[curArgIndex++], out x1);
+
+            if (curArgIndex >= l)
+                goto finishInitialization;
+            if (!BlockUtils.TryGetFloat(x[curArgIndex++], out rtol))
+                throw new Exception("Invalid value");
+
+            if (curArgIndex >= l)
+                goto finishInitialization;
+            if (!BlockUtils.TryGetBool(x[curArgIndex++], out full_output))
+                throw new Exception("Invalid value");
+
+            finishInitialization:
+            if (tol <= 0)
+                throw new Exception("tol too small (" + tol + " <= 0)");
+            if (maxiter < 1)
+                throw new Exception("maxiter must be greater than 0");
+            float p0 = x0;
+            int itr = 0;
+            float p = 0;
+            if (fprime is FuncCtx)
+            {
+                // Newton - Raphson method
+                for (; itr < maxiter; ++itr)
+                {
+                    // first evaluate fval
+                    float fval = (float)Block.CallFunc(ctx, func, new object[] { p0 });
```

Comment:
I think for performance it is better not to allocate object arrays on each iteration, but make var args = new object[1] once and then just rewrite it's args[0] = p0.
The same applies to all CallFunc calls.